pkg/kbagent/util/event.go

```diff
@@ -70,26 +73,40 @@ func createEvent(reason string, message string) *corev1.Event {
 			Component: proto.ProbeEventSourceComponent,
 			Host:      nodeName(),
 		},
-		FirstTimestamp:      metav1.Now(),
-		LastTimestamp:       metav1.Now(),
+		FirstTimestamp:      now,
+		LastTimestamp:       now,
 		EventTime:           metav1.NowMicro(),
 		ReportingController: proto.ProbeEventReportingController,
 		ReportingInstance:   podName(),
 		Action:              reason,
 		Type:                "Normal",
+		Count:               1,
 	}
 }
 
-func sendEvent(event *corev1.Event) error {
+func createOrUpdateEvent(event *corev1.Event) error {
 	clientSet, err := getK8sClientSet()
 	if err != nil {
 		return err
 	}
+	eventsClient := clientSet.CoreV1().Events(namespace())
 	for i := 0; i < sendEventMaxAttempts; i++ {
-		_, err = clientSet.CoreV1().Events(namespace()).Create(context.Background(), event, metav1.CreateOptions{})
-		if err == nil {
-			return nil
+		existingEvent, getErr := eventsClient.Get(context.Background(), event.Name, metav1.GetOptions{})
+		if getErr != nil {
+			_, createErr := eventsClient.Create(context.Background(), event, metav1.CreateOptions{})
+			if createErr == nil {
```

Comment:
save the createErr as the last error and return it when out of attempts.

Reply:
Done.